src/pages/ReimbursementAccount/NonUSD/Country/substeps/Confirmation.tsx

```diff
@@ -1,48 +1,114 @@
+        return ValidationUtils.getFieldRequiredErrors(values, [COUNTRY]);
+    }, []);
+
+    useEffect(() => {
+        if (currency === CONST.CURRENCY.EUR) {
+            return;
+        }
+
+        FormActions.setDraftValues(ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM, {[COUNTRY]: currencyMappedToCountry});
+        setSelectedCountry(currencyMappedToCountry);
+    }, [currency, currencyMappedToCountry]);
+
     return (
         <SafeAreaConsumer>
             {({safeAreaPaddingBottomStyle}) => (
                 <ScrollView
                     style={styles.pt0}
                     contentContainerStyle={[styles.flexGrow1, safeAreaPaddingBottomStyle]}
                 >
+                    <Text style={[styles.textHeadlineLineHeightXXL, styles.ph5, styles.mb3]}>{translate('countryStep.confirmBusinessBank')}</Text>
+                    <MenuItemWithTopDescription
+                        description={translate('common.currency')}
+                        title={currency}
+                        interactive={false}
+                    />
+                    <Text style={[styles.ph5, styles.mb3, styles.mutedTextLabel]}>
+                        {`${translate('countryStep.yourBusiness')} ${translate('countryStep.youCanChange')}`}
+                        {` `}
+                        <PressableWithoutFeedback
+                            accessibilityRole="button"
+                            accessibilityLabel={translate('common.settings')}
+                            accessible
+                            onPress={handleSettingsPress}
+                        >
+                            <Text style={[styles.label, styles.textBlue]}>{translate('common.settings').toLowerCase()}</Text>
+                        </PressableWithoutFeedback>
+                        .
+                    </Text>
                     <FormProvider
                         formID={ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM}
                         submitButtonText={translate('common.confirm')}
+                        validate={validate}
                         onSubmit={onNext}
                         style={[styles.flexGrow1]}
                         submitButtonStyles={[styles.mh5, styles.pb0, styles.mbn1]}
+                        isSubmitDisabled={disableSubmit}
                     >
-                        <Text style={[styles.textHeadlineLineHeightXXL, styles.ph5, styles.mb3]}>{translate('countryStep.confirmBusinessBank')}</Text>
-                        {/* This is only to showcase usage of PushRowWithModal component. The actual implementation will come in next issue - https://github.com/Expensify/App/issues/50897 */}
-                        <PushRowWithModal
-                            optionsList={CONST.ALL_COUNTRIES}
+                        <InputWrapper
+                            InputComponent={PushRowWithModal}
+                            optionsList={shouldAllowChange ? CONST.ALL_EUROPEAN_COUNTRIES : CONST.ALL_COUNTRIES}
```

Comment:
Is this necessary? Because I saw below that we only allow user to change the value when `shouldAllowChange` is true.
```suggestion
                            optionsList={CONST.ALL_EUROPEAN_COUNTRIES}
```

Reply:
Yes. If we only provide European countries here then we cant show Australia or Canada